### ISSUE TITLE
Library (Auth UI) does not work with SDK34 or higher (IllegalArgumentException)

### ISSUE THREAD

**Issue by dhoehl** [firebase/FirebaseUI-Android#2157]
Welcome to FirebaseUI and thanks for submitting an issue!

Please take a look at [open issues](https://github.com/firebase/FirebaseUI-Android/issues?q=is%3Aopen+is%3Aissue), as well as [resolved issues](https://github.com/firebase/FirebaseUI-Android/issues?q=is%3Aissue+is%3Aclosed), to see if your issue is either already being addressed, or has been solved by someone else.

If not, please feel free to fill in the following info so we can help faster!

### Step 1: Are you in the right place?

  * For issues or feature requests related to the code **in this repository** file a GitHub issue.
  * For general technical questions, post a question on [StackOverflow](http://stackoverflow.com/) tagged appropriately.
  * For general Firebase discussion, use the [firebase-talk google group](https://groups.google.com/forum/#!forum/firebase-talk)
  * For help troubleshooting your application that does not fall under one of the above categories, reach out to the [personalized Firebase support channel](https://firebase.google.com/support/contact/troubleshooting/)

### Step 2: Describe your environment

  * Android device: _____
  * Android OS version: SDK34
  * Google Play Services version: _____
  * Firebase/Play Services SDK version: _____
  * FirebaseUI version: 8.0.2
  
### Step 3: Describe the problem:

"java.lang.IllegalArgumentException: Targeting U+ (version 34 and above) disallows creating or retrieving a PendingIntent with FLAG_MUTABLE, an implicit Intent within and without FLAG_NO_CREATE and FLAG_ALLOW_UNSAFE_IMPLICIT_INTENT for security reasons. To retrieve an already existing PendingIntent, use FLAG_NO_CREATE, however, to create a new PendingIntent with an implicit Intent use FLAG_IMMUTABLE."

#### Steps to reproduce:

Compile with U or higher
  
#### Observed Results:

Crash
  
#### Expected Results:

No crash

**Comment by awesome-bhatti** [comment]
I am facing the same issue, it seems this library is not maintained anymore

**Comment by dhoehl** [comment]
This is bad. Is there an alternative? Even the google docs reference this library. Hope we get some offical answer as on how to move on. This is a braking change now. Otherwise I think we can fork this lib and fix it. There should be plenty of users out there

**Comment by ArcherEmiya05** [comment]
May I know which IdP you are using? It is working on my end using emulator with only Phone Auth.

**Comment by dhoehl** [comment]
> May I know which IdP you are using? It is working on my end with only Phone Auth.

eMail

**Comment by dhoehl** [comment]
This indeed fixes the problem for now, But I also could only use 20.7.0. With 21.0.0 the whole library crashes instantly. I won't close this issue, as it is only a workaround solution. The library itself should include the play-services dependency and handle the version correctly.

**Comment by k0taro9** [comment]
Hey, I've never used FirebaseUI before, so i have no idea what to import when using the latest versions. 
The implementation syntax has changed and some sources say to include
implementation(platform("com.google.firebase:platform-bom:33.0.2"))
others say nothing about using "platform" or "bom".
Google hasn't updated any of their documentation and nothing makes sense.
I just want to use email/password auth and google sign in with the ui element provided by this library.
Help would be very much appreciated.

**Comment by paulotaylor** [comment]
Yeah, it does look like this project is very much stale nowadays. Nothing wrong with that, but it's pretty bad that Firebase still mentions Firebase UI in their documentation while it's obvious that the new version of the play-services-auth library completely breaks this.

**Comment by ian-sayles** [comment]
The iOS version is still being maintained so hopefully this will be updated soon.

**Comment by mihaibarbulescu** [comment]
I can confirm that the issue reproduces at my side as well.

Works on com.google.android.gms:play-services-auth:20.7.0
Crashes on com.google.android.gms:play-services-auth:21.1.0

If 21.1.0 is used, results Gradle sync issue:
Failed to resolve: com.google.android.gms:play-services-auth:21.1.0

**Comment by ShaiAlkoby** [comment]
@thatfiredev any update on this? is anyone working on it or should we remove this dependency and implement it ourselves?

**Comment by hkchakladar** [comment]
Facing same issue. Looks like need to move on to other library or implement by ourselve.

Anyone have another library recommendation for phone auth only?

**Comment by dhoehl** [comment]
@thatfiredev It would be really nice if we at least get a short info if this is not maintained anymore. My app still works with the workaround of old play-services-auth version but eventually I need to implemented everything myself. Before I do this it would be nice to get the message that you won't work on this anymore

**Comment by thatfiredev** [comment]
Hi all,
Sorry for the long delay on this one! We finally released [version 9.0.0](https://github.com/firebase/FirebaseUI-Android/blob/master/docs/upgrade-to-9.0.md) of this library which fixes this issue. :)

I will close this issue for now, but please leave a comment or re-open if you're still facing this.
